airflow/api_fastapi/execution_api/routes/task_instances.py

```diff
@@ -391,6 +392,52 @@ def ti_update_state(
         )
 
 
+@router.patch(
+    "/{task_instance_id}/skip-downstream",
+    status_code=status.HTTP_204_NO_CONTENT,
+    # TODO: Add description to the operation
+    # TODO: Add Operation ID to control the function name in the OpenAPI spec
+    # TODO: Do we need to use create_openapi_http_exception_doc here?
+    responses={
+        status.HTTP_404_NOT_FOUND: {"description": "Task Instance not found"},
+        status.HTTP_422_UNPROCESSABLE_ENTITY: {"description": "Invalid payload for the state transition"},
+    },
+)
+def ti_skip_downstream(
+    task_instance_id: UUID, ti_patch_payload: TISkippedDownstreamTasksStatePayload, session: SessionDep
+):
+    ti_id_str = str(task_instance_id)
+    now = timezone.utcnow()
+    tasks = ti_patch_payload.tasks
+
+    dag_id, run_id = session.execute(select(TI.dag_id, TI.run_id).where(TI.id == ti_id_str)).fetchone()
+
+    if isinstance(tasks[0], tuple):
+        query = (
+            update(TI)
+            .where(TI.dag_id == dag_id, TI.run_id == run_id, tuple_(TI.task_id, TI.map_index).in_(tasks))
+            .values(state=TaskInstanceState.SKIPPED, start_date=now, end_date=now)
+            .execution_options(synchronize_session=False)
+        )
+    else:
+        query = (
+            update(TI)
+            .where(TI.dag_id == dag_id, TI.run_id == run_id, TI.task_id.in_(tasks))
+            .values(state=TaskInstanceState.SKIPPED, start_date=now, end_date=now)
+            .execution_options(synchronize_session=False)
+        )
```

Comment:
is it possible to skip some mapped indexes and some non mapped tasks at the same time?

If so id suggest something like this
```suggestion
    task_ids = [
        task if isinstance(task, tuple) else (task, -1)
        for task in tasks
    ]

    query = (
        update(TI)
        .where(TI.dag_id == dag_id, TI.run_id == run_id, tuple_(TI.task_id, TI.map_index).in_(task_ids))
        .values(state=TaskInstanceState.SKIPPED, start_date=now, end_date=now)
        .execution_options(synchronize_session=False)
        )
```